tools/src/wyvern/target/corewyvernIL/type/ValueType.java

```diff
@@ -6,6 +6,7 @@
 import wyvern.target.corewyvernIL.support.View;
 
 public abstract class ValueType extends CaseType implements EmitOIR {
+	private boolean resourceFlag = false;
```

Comment:
Avoid adding a variable to this class (especially a mutable variable).  Instead just override isResource() in subclasses.